src/modals/SettingsModal/GeneralSettingsSection.tsx

```diff
@@ -114,75 +117,103 @@ const GeneralSettingsSection = ({ className }: GeneralSettingsSectionProps) => {
   }
 
   const discreetModeText = t('Discreet mode')
+
   const themeOptions = [
     { label: t('System'), value: 'system' as ThemeSettings },
     { label: t('Light'), value: 'light' as ThemeSettings },
     { label: t('Dark'), value: 'dark' as ThemeSettings }
   ]
 
+  // Make sure to find the closest available locktime if the previous one was set before using a Select component.
+  const currentLockTime = walletLockTimeInMinutes
+    ? locktimeInMinutes.reduce((prev, curr) =>
+        Math.abs(curr - walletLockTimeInMinutes) < Math.abs(prev - walletLockTimeInMinutes) ? curr : prev
+      )
+    : 0
+
   return (
     <Box className={className}>
       <KeyValueInput
-        label={t`Lock time`}
-        description={t`Duration in minutes after which an idle wallet will lock automatically.`}
+        label={t('Lock time')}
+        description={t('Duration in minutes after which an idle wallet will lock automatically.')}
         InputComponent={
-          <Input
+          <Select
             id="wallet-lock-time-in-minutes"
-            value={walletLockTimeInMinutes || ''}
-            onChange={(v) => handleWalletLockTimeChange(v.target.value)}
-            label={walletLockTimeInMinutes ? t`Minutes` : t`Off`}
-            type="number"
-            step={1}
-            min={1}
+            options={locktimeInMinutes.map((v) => ({
+              label: v ? `${v} ${t('Minutes')}` : t('Off'),
```

Comment:
nit: how about we lowercase the `Minutes` word? No need to add a new translation entry, we can achieve it with `.toLowerCase()`

Reply:
Same for below

Reply:
Yeah, I followed Ledger live on this one 🤔

Reply:
ok, no problem

Reply:
But I agree, I also prefer using lowercase. Updating..

Reply:
Mhh actually, using `.toLowerCase()` doesn't work because of language like German, where nouns starts with an upper case.

Reply:
Very valid point, I should have thought about this 😆